test/AxelarGatewaySinglesig.js

```diff
@@ -1320,6 +1226,252 @@ describe('AxelarGatewaySingleSig', () => {
+        await token.mint(contract.address, amount);
+
+        const deployTokenData = arrayify(
+          defaultAbiCoder.encode(
+            ['uint256', 'uint256', 'bytes32[]', 'string[]', 'bytes[]'],
+            [
+              CHAIN_ID,
+              ROLE_OWNER,
+              [getRandomID()],
+              ['deployToken'],
+              [
+                defaultAbiCoder.encode(
+                  ['string', 'string', 'uint8', 'uint256', 'address'],
+                  [name, symbol, decimals, capacity, token.address],
+                ),
+              ],
+            ],
+          ),
+        );
+
+        const salt = randomBytes(32);
+        const depositHandlerAddress = getCreate2Address(
+          contract.address,
+          salt,
+          keccak256(`0x${DepositHandler.bytecode}`),
+        );
+        await token
+          .connect(nonOwnerWallet)
+          .transfer(depositHandlerAddress, amount);
+
+        await getSignedExecuteInput(deployTokenData, ownerWallet).then(
+          (input) =>
+            expect(contract.execute(input))
+              .to.emit(contract, 'TokenDeployed')
+              .withArgs(symbol, token.address),
+        );
+
+        const getBurnTokenData = () =>
+          arrayify(
+            defaultAbiCoder.encode(
+              ['uint256', 'uint256', 'bytes32[]', 'string[]', 'bytes[]'],
+              [
+                CHAIN_ID,
+                ROLE_OWNER,
+                [getRandomID()],
+                ['burnToken'],
+                [defaultAbiCoder.encode(['string', 'bytes32'], [symbol, salt])],
+              ],
+            ),
+          );
+
+        const getMintTokenData = () =>
+          arrayify(
+            defaultAbiCoder.encode(
+              ['uint256', 'uint256', 'bytes32[]', 'string[]', 'bytes[]'],
+              [
+                CHAIN_ID,
+                ROLE_OWNER,
+                [getRandomID()],
+                ['mintToken'],
+                [
+                  defaultAbiCoder.encode(
+                    ['string', 'address', 'uint256'],
+                    [symbol, ownerWallet.address, amount],
+                  ),
+                ],
+              ],
+            ),
+          );
+
+        for (const adminWallet of adminWallets.slice(0, 3)) {
+          await contract.connect(adminWallet).freezeToken(symbol);
+        }
+
+        await getSignedExecuteInput(getBurnTokenData(), ownerWallet).then(
+          (input) =>
+            expect(contract.execute(input))
+              .not.to.emit(token, 'Transfer')
+              .withArgs(depositHandlerAddress, contract.address, amount),
+        );
+
+        await getSignedExecuteInput(getMintTokenData(), ownerWallet).then(
+          (input) =>
+            expect(contract.execute(input)).not.to.emit(token, 'Transfer'),
+        );
+
+        for (const adminWallet of adminWallets.slice(0, 3)) {
+          await contract.connect(adminWallet).unfreezeToken(symbol);
+        }
```

Comment:
```suggestion
        await Promise.all(
          adminWallets
            .slice(0, 3)
            .map((admin) => contract.connect(admin).unfreezeToken(symbol)),
        );
```
Doing `for` loop in js is almost never a good idea. It's either `.forEach` if sync or `Promise.all` if async. `Promise.all` is also much more performant than `for` loop if async.

Reply:
@sammy1991106  the problem with `Promise.all` or `.forEach` is that it will execute all 3 calls immediately and this might break the TX nonce. `for` loop with async with sequential performance is here on purpose, so we can wait for each call to finish.

Reply:
No it won't break nonce. Each admin is a different address.